internal/storage/storage.go

```diff
@@ -40,13 +60,25 @@ func (s *storage) Set(key string, value any) error {
 	if callback != nil {
 		callback(key, value)
 	}
+
+	err := s.jsonStorage.Set(key, value)
+	if err != nil {
+		return err
+	}
 	return nil
 }
 
 func NewStorage(opts ...storageOption) StorageWithCallbacks {
+	// we ignore the error and just work without locking
+	file, err := xdg.ConfigFile("snyk/ls-config")
+	if err != nil {
+		panic(err)
+	}
+
 	s := &storage{
-		data:      make(map[string]any),
-		callbacks: make(map[string]StorageCallbackFunc),
+		data:        make(map[string]any),
+		callbacks:   make(map[string]StorageCallbackFunc),
+		jsonStorage: configuration.NewJsonStorage(file),
```

Comment:
It might be ugly to call WithStorageFile(file)() here
but maybe we can extract the logic inside in smth like "initStorageFile" and use it here